gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/apim/core/api/use_case/UpdateFederatedApiUseCaseTest.java

```diff
@@ -206,4 +212,94 @@ public void update_creates_index() {
 
         assertThat(indexer.storage()).isNotEmpty().hasSize(1);
     }
+
+    @Test
+    void updating() {
+        ZonedDateTime inputDate = ZonedDateTime.now().plusDays(1);
+        Api input = Api
+            .builder()
+            .id("input")
+            .environmentId("input")
+            .crossId("input")
+            .name("input")
+            .description("input")
+            .version("input")
+            .originContext(new OriginContext.Integration("input"))
+            .definitionVersion(DefinitionVersion.V1)
+            .apiDefinitionV4(io.gravitee.definition.model.v4.Api.builder().build())
+            .apiDefinition(io.gravitee.definition.model.Api.builder().build())
+            .federatedApiDefinition(null)
+            .type(ApiType.MESSAGE)
+            .deployedAt(inputDate)
+            .createdAt(inputDate)
+            .updatedAt(inputDate)
+            .visibility(Api.Visibility.PRIVATE)
+            .lifecycleState(Api.LifecycleState.STOPPED)
+            .picture("input")
+            .groups(Set.of("input"))
+            .categories(Set.of("input"))
+            .labels(List.of("input"))
+            .disableMembershipNotifications(false)
+            .apiLifecycleState(Api.ApiLifecycleState.DEPRECATED)
+            .background("input")
+            .build();
```

Comment:
FYI, we have Fixtures classes like `fixtures.core.model.ApiFixtures` to help build models easily

Reply:
Yes but here I want set all fields exhaustively